src/django/api/reports.py

```diff
@@ -4,15 +4,35 @@
 from urllib.parse import quote
 
 from django.db import connection
-
+from api.models import Facility
 
 _root = os.path.abspath(os.path.dirname(__file__))
 
 
+def monthly_promoted_name_and_address():
+    data = dict()
+    fs = Facility.objects.all()
+    for f in fs:
+        hs = f.history.exclude(created_from_id__isnull=True)
+        for h in hs:
```

Comment:
Since we only have one reference to each of these QuerySet variables, consider moving them into the `for` definitions.

```
    for f in Facility.objects.all():
        for h in f.history.exclude(created_from_id__isnull=True):
```

Reply:
Updated in 44a9a1f